tokio/src/net/tcp/socket.rs

```diff
@@ -378,7 +378,68 @@ impl TcpSocket {
     ///
     /// [`set_linger`]: TcpSocket::set_linger
     pub fn linger(&self) -> io::Result<Option<Duration>> {
-        self.inner.get_linger()
+        self.inner.linger()
+    }
+
+    /// Gets the value of the `IP_TOS` option for this socket.
+    ///
+    /// For more information about this option, see [`set_tos`].
+    ///
+    /// **NOTE:** On Windows, `IP_TOS` is only supported on [Windows 8+ or
+    /// Windows Server 2012+.](https://docs.microsoft.com/en-us/windows/win32/winsock/ipproto-ip-socket-options)
+    ///
+    /// [`set_tos`]: Self::set_tos
+    // https://docs.rs/socket2/0.4.2/src/socket2/socket.rs.html#1178
+    #[cfg(not(any(
+        target_os = "fuchsia",
+        target_os = "redox",
+        target_os = "solaris",
+        target_os = "illumos",
+    )))]
+    #[cfg_attr(
+        docsrs,
+        doc(cfg(all(
+            unix,
+            not(any(
+                target_os = "fuchsia",
+                target_os = "redox",
+                target_os = "solaris",
+                target_os = "illumos",
+            ))
+        )))
+    )]
```

Comment:
One of these includes a `all(unix, ...)` and the other doesn't. Why?

Reply:
Good catch. I forgot to remove `all(unix, ...)` when I copied the doc cfg from `TcpSocket::reuseport`.